node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -119,7 +120,10 @@ object WebApi {
       blockApi.isFinalized(hash).flatMap(_.liftToBlockApiErr)
 
     def getTransaction(hash: String): F[TransactionResponse] =
-      cacheTransactionAPI.getTransaction(hash)
+      OptionT
+        .whenF(hash.nonEmpty)(cacheTransactionAPI.getTransaction(hash))
+        .value
+        .flatMap(_.liftTo(new BlockApiException("Block hash cannot be empty.")))
```

Comment:
I didn't realize this the first time, but now exception for empty hash will be thrown even for correct hash which is not available in transaction API.

I found pattern with Option#filter to be concise solution in these situations.
```scala
Option(hash)
  .filter(_.nonEmpty)
  .liftTo(new BlockApiException("Block hash cannot be empty."))
  .flatMap(cacheTransactionAPI.getTransaction)
```

Reply:
And for correct hash when `cacheTransactionAPI.getTransaction` returns None?

Reply:
WDY mean _correct hash_? `cacheTransactionAPI.getTransaction` cannot returns None because its returns  type is `F[TransactionResponse]`.